pennylane/_grad.py

```diff
@@ -75,7 +74,7 @@ def _get_grad_fn(self, args):
           inspecting as to which of the parameter arguments are marked
           as differentiable.
         """
-        if self._grad_fn is not None:
+        if self._grad_fn:
```

Comment:
How come this was changed?

Reply:
Ah my bad, maybe I shouldn't have. Is explicit comparison to None preferred to implicit Boolean conversion?

Reply:
It is context dependent, and I couldn't quite tell the context here, but in general yes --- if we are checking whether something is `None` or not, better to use `is (not) None` than to rely on the fact that `bool(None) == False`.

This is partly because allowing implicit boolean casting here can lead to the introduction of bugs; e.g., if `self._grad_fn` is for some reason `0`, this is `not None`, _but_ it is also `False`! So the wrong branch will be taken.

Reply:
I think it would be very bad if `_grad_fn` was 0 since it will be called, but I will change it back :)